File: Number_Guessing_Games.dart
import 'dart:io';
import 'dart:math';

void main() {
  final Random random = Random();
  final int secretNumber = random.nextInt(100) + 1;
  int attempts = 0;

  print("Bem-vindo ao jogo de adivinhação!");

  while (true) {
    stdout.write("Digite um número: ");
    final int guess = int.tryParse(stdin.readLineSync()!) ?? 0;
    attempts++;

    if (guess == secretNumber) {
      print("Parabéns! Você acertou em $attempts tentativas.");
      break;
    } else if (guess < secretNumber) {
      print("Tente um número maior.");
    } else {
      print("Tente um número menor.");
    }
  }
}

File: Number_Guessing_Games.java
import java.util.Scanner;

public class GuessTheNumber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int secretNumber = (int) (Math.random() * 100) + 1;
        int attempts = 0;

        System.out.println("Bem-vindo ao jogo de adivinhação!");

        while (true) {
            System.out.print("Digite um número: ");
            int guess = scanner.nextInt();
            attempts++;

            if (guess == secretNumber) {
                System.out.println("Parabéns! Você acertou em " + attempts + " tentativas.");
                break;
            } else if (guess < secretNumber) {
                System.out.println("Tente um número maior.");
            } else {
                System.out.println("Tente um número menor.");
            }
        }

        scanner.close();
    }
}
